sdk/core/Azure.Core/src/Pipeline/Internal/TelemetryPolicy.cs

```diff
@@ -24,7 +24,10 @@ public TelemetryPolicy(string componentName, string componentVersion, string? ap
 
         public override void OnSendingRequest(HttpMessage message)
         {
-            message.Request.Headers.Add(HttpHeader.Names.UserAgent, _header);
+            var header = _header;
+            if (message.TryGetProperty("SDKUserAgent", out var userAgent) && userAgent is string userAgentString)
```

Comment:
Could we call the property "ApplicationId" and instead of treating it as an alternative header, we treat it as an alternative (per-request) ApplicationId.

Reply:
Yeah if we go this route we can definitely name it whatever we want.  There might be some ambiguity with application id and the full user-agent string constructed from the application id that we want to consider here.